code/game/objects/structures/nest.dm

```diff
@@ -0,0 +1,66 @@
+// Nests spawn mobs and objects when crossed by players.
+// Nests placed in "spawn_trigger_distance" will always spawn the same type of mob.
+
+/obj/structure/nest
+	name = "tunnel"
+	desc = "A twisted, dark passage to the underground."
+	icon = 'icons/mob/nest.dmi'
+	icon_state = "hole"
+
+	move_resist = INFINITY
+	anchored = TRUE
+	density = FALSE
+
+	var/faction = list("hostile")
+	var/spawn_byproduct = list(/obj/item/stack/ore/glass, /obj/item/stack/ore/iron)
+	var/spawn_byproduct_max = 3
+	var/spawn_is_triggered = FALSE
+	var/spawn_max = 2
+	var/spawn_mob_options = list(/mob/living/simple_animal/crab)
+	var/spawn_trigger_distance = 7
```

Comment:
Could you add some autodoc comments for these variables?

Reply:
Added in 0b4b3a490292d14267a3917a7e4461defaa87072, also a summary on the top of the file. Please let me know if something is badly worded!